packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeFormatService.ts

```diff
@@ -0,0 +1,566 @@
+      mimeType: 'application/json',
+      data: new AttachmentData({
+        json: JsonTransformer.toJSON(presentationExchangeRequestMessage),
+      }),
+    })
+
+    return { format, attachment }
+  }
+  public async processRequest(options: ProcessRequestOptions): Promise<void> {
+    if (!options.requestAttachment) {
+      throw Error('Request message is missing while processing proof request in presentation exchange.')
+    }
+
+    const requestMessage = options.requestAttachment
+
+    const requestPresentation = requestMessage.attachment.getDataAsJson<RequestPresentationExchangeOptions>()
+
+    const pex: PEXv1 = new PEXv1()
+    const result: Validated = pex.validateDefinition(requestPresentation.presentationDefinition)
+
+    if (Array.isArray(result) && result[0].status !== Status.INFO) {
+      throw new AriesFrameworkError(
+        `Error in presentation definition while processing presentation exchange request: ${result[0].message} `
+      )
+    }
+  }
+
+  public async createPresentation(
+    agentContext: AgentContext,
+    options: FormatCreatePresentationOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofAttachmentFormat> {
+    if (!options.proofFormats.presentationExchange) {
+      throw Error('Presentation Exchange format missing while creating presentation in presentation exchange service.')
+    }
+
+    const requestPresentation = options.attachment.getDataAsJson<RequestPresentationExchangeOptions>()
+
+    const credential: IVerifiableCredential = options.proofFormats.presentationExchange.formats
+
+    const pex: PEXv1 = new PEXv1()
+
+    const subject: ICredentialSubject = credential.credentialSubject as ICredentialSubject
+
+    // Credential is allowed to be presented without a subject id. In that case we can't prove ownerhsip of credential
+    // And it is more like a bearer token.
+    // In the future we can first check the holder key and if it exists we can use that as the one that should authenticate
+    // https://www.w3.org/TR/vc-data-model/#example-a-credential-issued-to-a-holder-who-is-not-the-only-subject-of-the-credential-who-has-no-relationship-with-the-subject-of-the-credential-but-who-has-a-relationship-with-the-issuer
+    if (!subject?.id) {
+      throw new AriesFrameworkError(
+        'Credential subject missing from the selected credential for creating presentation.'
+      )
+    }
+
+    const didResolutionResult = await this.didResolver.resolve(agentContext, subject.id)
+
+    if (!didResolutionResult.didDocument) {
+      throw new AriesFrameworkError(`No did document found for did ${subject.id}`)
+    }
+
+    if (!didResolutionResult.didDocument?.authentication) {
+      throw new AriesFrameworkError(`No did authentication found for did ${subject.id} in did document`)
+    }
+
+    if (!didResolutionResult.didDocument?.verificationMethod) {
+      throw new AriesFrameworkError(`No did verification method found for did ${subject.id} in did document`)
+    }
+
```

Comment:
A document doesn't necearily have the `verificationMethod` property. I think we should remove these checks and let it handle by other methods (see next comments)